internal/app/singularity/remote_add.go

```diff
@@ -10,12 +10,21 @@ import (
 	"net/url"
 	"os"
 	"path"
+	"strings"
 
 	"github.com/sylabs/singularity/internal/pkg/remote"
 )
 
 // RemoteAdd adds remote to configuration
 func RemoteAdd(configFile, name, uri string, global bool) (err error) {
+	// Explicit handling of corner cases: name and uri must be valid strings
+	if strings.TrimSpace(name) == "" {
+		return fmt.Errorf("invalid name: %s", name)
```

Comment:
Printing an error with an empty string might be a bit confusing. How about something like `invalid name: cannot have empty name`

Reply:
Good point, fixed